docs/readthedocs/source/doc/PPML/Overview/ali_ecs_occlum_cn.md

```diff
@@ -0,0 +1,509 @@
+ 
+
+图 PPML基本架构
+
+## 1. 环境部署
+
+以下以阿里云环境为例，如果是基于裸金属机器搭建，请参考附录。
+首先，我们需要一个安装了SGX Plugin的K8S集群环境。在本例中，我们在阿里云申请了两台g7t的ECS实例（ecs.g7t.4xlarge），基本配置如下。
+| CPU核数  | 内存 | 安全内存（EPC）| 操作系统 |
+| ------------- | ------------- | ------------- | ------------- |
+| 32  | 64GB  | 32GB | Ubuntu 20.04 LTS 2 |
+
+用户也可以根据自己的需求申请不同配置的ECS安全实例。
+VM OS选择Ubuntu20.04 LTS2, 这也是Occlum标准release所基于的操作系统。
+另外，系统内核需要升级以支持SGX。
+
+```bash
+sudo apt install --install-recommends linux-generic-hwe-20.04
+```
+
+然后，需要在每台实例上配置安装K8S环境，并配置安装K8S SGX plugin。
+细节不再赘述，用户可以参考技术文档《在K8S上部署可扩展的基于Occlum的安全推理实例》或者附录的相关部分。
+
+## 2. 快速上手
+
+本章会介绍PPML基本概念，以及如何用BigDL PPML occlum image在SGX中执行应用程序。
+需要注意的是：为了简化上手流程，我们会在运行环境中编译和运行SGX enclave；这种运行方式会有安全风险，仅能用于开发和测试，实际部署需要参照后面的生产环境部署章节。
+
+### 2.1 基本概念
+
+SGX应用需要编译(build)成SGX enclave，才能加载到SGX中运行。通常情况下，开发人员需要用SGX SDK重新编写应用，才能编译成合法的enclave，但这样的开发代价较大，维护成本也较高。为了避免上述问题，我们可以用Occlum实现应用的无缝迁移。Occlum是为SGX开发的LibOS应用，它可以将应用的系统调用翻译成SGX可以识别的调用，从而避免修改应用。BigDL PPML在Occlum的基础上，又进行了一次封装和优化，使得大数据应用，如Spark/Flink可以无缝的运行在SGX上。
+ 
+图 SGX enclave结构
+作为硬件级的可信执行环境，SGX的攻击面非常小，攻击者即使攻破操作系统和BIOS也无法获取SGX中的应用和数据。但在端到端的应用中，用户还需要确保其他阶段的安全性。简而言之，用户需要确保数据或者文件在SGX外部是加密的，仅在SGX内部被解密和计算，如下图所示。为了实现这个目的，我们往往需要借助密钥管理服务 (Key Management Service, KMS) 的帮助。用户可以将密钥托管到KMS，等应用在SGX中启动后，再从KMS申请和下载密钥。
+ 
+图 SGX应用设计原则
+PPML项目的核心功能是帮助用户迁移现有的应用，用户可以选择迁移现有的大数据AI应用，也可以选择开发全新的应用。PPML应用的开发和常规应用基本相同。例如PySpark的应用代码和常规应用并没有区别。但在设计、编译和部署时有一定的差异。具体表现为：
+* 设计时需要考虑加解密流程，确保明文数据只出现在SGX内部
+* 编译时，需要通过Occlum将应用编译成SGX enclave
+* 部署时，需要将SGX enclave部署到有SGX环境的节点
+在剩下的章节中，我们以PySpark运行SQL和sklearn求线性回归方程为例，介绍如何
+* 通过docker部署单机PySpark应用。
+* 通过K8S部署分布式PySpark应用。
+
+前者主要针对小数据量的单机环境，后者主要针对大数据量的分布式环境。
+
+### 2.2 PySpark执行SQL任务
+
+SparkSQL是Spark生态中的核心功能之一。通过Spark提供的SQL接口，数据分析师和开发人员可以通撰写简单的SQL语句实现对TB/PB级别数据的高效查询。在下面的例子中，我们将介绍如何通过Python格式的SQL文件，查询大规模数据。
+
+#### 2.2.1 部署运行在docker容器中
+1.	配置合适的资源，启动运行脚本`start-spark-local.sh`进入docker image中。
+
+```bash
+# Clean up old container
+sudo docker rm -f bigdl-ppml-trusted-big-data-ml-scala-occlum
+ 
+# Run new command in container
+sudo docker run -it --rm \
+--net=host \
+--name=bigdl-ppml-trusted-big-data-ml-scala-occlum \
+--cpuset-cpus 10-14 \
+--device=/dev/sgx/enclave \  #需提前配置好的sgx环境
+--device=/dev/sgx/provision \
+-v /var/run/aesmd:/var/run/aesmd \
+-v data:/opt/occlum_spark/data \
+-e SGX_MEM_SIZE=24GB \   #EPC即使用的SGX内存大小
+-e SGX_THREAD=1024 \
+-e SGX_HEAP=1GB \
+-e SGX_KERNEL_HEAP=1GB \
+-e SGX_LOG_LEVEL=off \intelanalytics/bigdl-ppml-trusted-big-data-ml-scala-occlum:2.2.0-SNAPSHOT \
+bash
+2.	编写python源码，如sql_example.py 并将其放置在image的目录py-examples下
+3.	修改/opt/run_spark_on_occlum_glibc.sh文件，设置程序启动入口。
+run_pyspark_sql_example() {
+    init_instance spark  #执行occlum init初始化occlum文件结构并设置对应配置
+    build_spark #拷贝依赖并执行occlum build 构建可执行程序
+    cd /opt/occlum_spark
+    echo -e "${BLUE}occlum run pyspark SQL example${NC}"
+    occlum run /usr/lib/jvm/java-8-openjdk-amd64/bin/java \
+                -XX:-UseCompressedOops -XX:MaxMetaspaceSize=$META_SPACE \
```

Comment:
MaxMetaspaceSize is removed now，the two examples need to update.

Reply:
Thank you for your comments! 
MaxMetaspaceSize removed from examples.